src/ngx_http_lua_kong_log_handler.c

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2019-2023 Kong Inc.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *    http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ngx_http_lua_kong_common.h"
+
+
+/* This function contains the logic to append the Request ID to the error log */
+static u_char *
+ngx_http_lua_kong_error_log_handler(ngx_http_request_t *r, u_char *buf, size_t len)
+{
+    ngx_http_variable_value_t    *value;
+    ngx_http_lua_kong_loc_conf_t *lcf;
+
+    lcf = ngx_http_get_module_loc_conf(r, ngx_http_lua_kong_module);
+    if (lcf->req_id_var_index == NGX_CONF_UNSET) {
+        return buf;
+    }
+
+    value = ngx_http_get_indexed_variable(r, lcf->req_id_var_index);
+    if (value == NULL || value->not_found) {
+        return buf;
+    }
+
+    buf = ngx_snprintf(buf, len, ", kong_request_id: %v", value);
+    return buf;
+}
+
+
+/*
+ * This function is a replacement for the original nginx error
+ * log handler (r->log_handler). It executes the original logic
+ * and then our error log handler: ngx_http_lua_kong_error_log_handler
+ */
+static u_char *
+ngx_http_lua_kong_combined_error_log_handler(ngx_http_request_t *r, ngx_http_request_t *sr,
+    u_char *buf, size_t len)
+{
+    u_char                       *p;
+    ngx_http_lua_kong_ctx_t      *ctx;
+
+    ctx = ngx_http_lua_kong_get_module_ctx(r);
+    if (ctx != NULL) {
```

Comment:
I have a question: 
if we don't set the directive `lua_kong_error_log_request_id`,
do we pass the original log handler calling?
Does it means we will not log the normal http message?

Reply:
no we don't skip the original log handler calling, when the directive is unset we still execute the combined error log handler but the `ngx_http_lua_kong_error_log_handler` [has no effect](https://github.com/Kong/lua-kong-nginx-module/blob/2def18c97bc1521d012f1cff714ddf707f887110/src/ngx_http_lua_kong_log_handler.c#L29) since there is no request id variable index set.

I added [a check in the sanity test case](https://github.com/Kong/lua-kong-nginx-module/blob/ed6732b9af5f8c1ac06dfc81b37e18660a649551/t/009-error-log-append.t#L35) to verify this.